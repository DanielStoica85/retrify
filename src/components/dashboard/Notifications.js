import React from 'react';

const Notifications = () => {
    return (
        <div>
            <p>Notifications</p>
            <p>
                You are running this application in{' '}
                <b>{process.env.NODE_ENV}</b>.
            </p>
        </div>
    );
};

export default Notifications;
