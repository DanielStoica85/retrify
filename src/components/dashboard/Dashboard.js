import React, { Component } from 'react';
import { Container, Row, Col } from 'reactstrap';
import Notifications from './Notifications';
import RetrosList from '../retros/RetrosList';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class Dashboard extends Component {
    render() {
        return (
            <Container className="dashboard">
                <Row>
                    <Col sm="12" md="6">
                        <RetrosList retros={this.props.retros} />
                    </Col>
                    <Col sm="12" md={{ size: 5, offset: 1 }}>
                        <Notifications />
                    </Col>
                </Row>
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        retros: state.retros
    };
};

Dashboard.propTypes = {
    retros: PropTypes.array
};

export default connect(mapStateToProps)(Dashboard);
