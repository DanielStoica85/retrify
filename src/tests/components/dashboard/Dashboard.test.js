import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../../../components/dashboard/Dashboard';

describe('Dashboard component', () => {
    it('should render correctly', () => {
        const component = shallow(<Dashboard />);
        expect(component).toMatchSnapshot();
    });
});
