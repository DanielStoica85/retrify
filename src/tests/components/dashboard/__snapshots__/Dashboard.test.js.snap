// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render correctly 1`] = `
<Container
  className="dashboard"
  tag="div"
>
  <Row
    tag="div"
  >
    <Col
      md="6"
      sm="12"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <RetrosList />
    </Col>
    <Col
      md={
        Object {
          "offset": 1,
          "size": 5,
        }
      }
      sm="12"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Notifications />
    </Col>
  </Row>
</Container>
`;
